### DISCUSSION TITLE
My enhanced index.html for the server with more features (regen, edit, history etc.)

### DISCUSSION THREAD

**Discussion by Equim-chan** [ggml-org/llama.cpp#2777] [Show and tell]
## Code
https://gist.github.com/Equim-chan/1c46aca63990e1800c4a3f002b1e3aee

## Description
I initially wanted to submit a PR, however my modifications grew over time and in the end became a bit too much to merge. Despite this, I still want to share it for others who might find it useful or interesting.

![image](https://github.com/ggerganov/llama.cpp/assets/17795845/cad0df01-0c7f-4480-abaf-6b6d0dddf073)

### Menu
https://github.com/ggerganov/llama.cpp/assets/17795845/0ceb3e4e-1ff5-4e16-813c-8a2c557fa7fe

### Editing
https://github.com/ggerganov/llama.cpp/assets/17795845/0748cb92-7736-4bdb-a02e-9dff5f99321a

### History
https://github.com/ggerganov/llama.cpp/assets/17795845/0798f85a-21eb-432e-bf45-2b23abc3ad90

**Comment by ggerganov** [comment]
I really like it! I personally don't think there is a problem to merge this, though I'll leave the decision up to the people involved in the server development

**Comment by SlyEcho** [comment]
I will try to merge it.

**Comment by staviq** [comment]
@SlyEcho In case you have trouble merging it with recent template autosave functionality, let me know.

**Comment by ghost** [comment]
Presets is a great idea & `Edit Last` from the 2nd clip is amazing.

**Comment by kroryan** [comment]
im trying to use it i create an index and copy paste it and i did add the text to completion.js but the web doesnt load its just blank, i think im doing something wron with the completion.js



okey so it was because i was adding it to the begining of completion.js



pd: how can i save the conversation as json?
